### ISSUE TITLE
[Feature request?] Alternative files with submodules

### ISSUE THREAD

**Issue by pschmitt** [TheLocehiliosan/yadm#78]
Hi,
I've been experimenting with yadm and was wondering whether I've been doing something wrong.

So I have most of my config files in their own repos and as such I added them as submodules. 
But this way alternates files are not processed, right? Do I have to add them directly to the yadm repo?

**Comment by pschmitt** [comment]
So I went ahead and checked. In case someone was wondering the same: YES, you need to have yadm track your files directly for the alt feature to work.
In its current state it walks over all the tracked files, via `git ls-files` which does not list the files inside submodules.

**Comment by TheLocehiliosan** [comment]
@pschmitt Yes, your assessment of the current functionality is correct. I will consider including files from submodules for alternate file processing.

**Comment by tox2ik** [comment]
@pschmitt Hi, I want alt-files to be supported in sub-modules as well. Your suggestion (`git submodule foreach 'git ls-files | sed "s|^|$path/|"''`) seems to do the trick. However, given some previous commits in the history, one goal fo the project seems to be bash-only, hence sed is an external tool and should not be used. Be that as it may...

I am having diffuclties understanding your point about .gitignore in submodules. Can you please elaborate on why not-considering submodule-.gitignore would pose problems? In general I don't see the issue. 

It could be an issue if one uses "git add ." in the submodule, because the parent yadm project has created new symlinks. TBH, I don't think this is problematic enough to warrant any more complicated handling of submodules .gitignore. My stance being; when you git add, you should be careful.
